### PR TITLE
Update `Remove`/`RemoveRange`/`RemoveAll` calls in line with framework changes

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on ppy/osu-framework#5382

### CODE REVIEW

File: osu.Game/Graphics/Containers/SectionsContainer.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public Drawable ExpandableHeader
                 if (value == expandableHeader) return;
 
                 if (expandableHeader != null)
-                    RemoveInternal(expandableHeader);
+                    RemoveInternal(expandableHeader, false);
```
Comment (smoogipoo) [inline osu.Game/Graphics/Containers/SectionsContainer.cs:38]: Any special reason for this one to be `false` when the one below is `true`?

**Review by smoogipoo — DISMISSED** [review]
Looks alright, super hard to review though.

**Review by frenzibyte — DISMISSED** [review]

**Comment by frenzibyte** [comment]
Test failures seem relevant, `CheckAudioQualityTest.getMockWorkingBeatmap` would probably have to be adjusted since `Track` now requires an argument on constructor.

File: osu.Game/Graphics/Containers/SectionsContainer.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Graphics/Containers/SectionsContainer.cs:38]: Was an oversight, fixed.

**Comment by frenzibyte** [comment]
Turns out `UserProfileOverlay` recreates the sections container when switching between users (to handle hiding certain sections on some profiles), and removes the header before disposing the old sections container to reuse it in the new container:

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Overlays/UserProfileOverlay.cs#L54-L55

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Overlays/UserProfileOverlay.cs#L81-L91

I've went ahead and removed disposing from both `ExpandableHeader` and `Footer` for simplicity.

**Review by frenzibyte — APPROVED** [review]

**Pull request merged**